TACC/Core-Portal
# FP-990 & FP-1255: Update <IntroMessage> to Use Dismissible <SectionMessage>

## To Do

<details><summary>Done</summary>

- [x] Reduce harshness of section message text and border (or create new ticket). — [source](https://tacc-team.slack.com/archives/CQUA4D5KJ/p1650401626612759)
- [x] Get Design approval on proposed new colors. — [request](https://tacc-team.slack.com/archives/CQUA4D5KJ/p1650468639024599?thread_ts=1650401626.612759&cid=CQUA4D5KJ)

</details>

## Overview: ##

Use `SectionMessage`, not `Alert`, for page welcome messages.

## Related Jira tickets: ##

* [FP-990](https://jira.tacc.utexas.edu/browse/FP-990)
* [FP-1255](https://jira.tacc.utexas.edu/browse/FP-1255)

## Summary of Changes: ##

- Support dismissible `<SectionMessage>`
- Use `<SectionMessage>` not `<Alert>`.
- Update documentation and tests.
- Tweak CSS.

## Testing Steps: ##

FP-990
1. Follow https://github.com/TACC/Core-Portal/pull/191 "Testing Steps".
2. Confirm message design matches `<SectionMessage>` (see UI Patterns) not `<Alert>`.

FP-1225
1. Open [UI Patterns](https://cep.dev/workbench/ui-patterns).
2. Scroll to Messages & Notifications.
3. Dismiss the Section messages in the content—confirm they disappear.
4. Reload the section—confirm the dismissed Section messages have returned.

## UI Photos:

| Before | Initial UI | [New Approved UI](https://tacc-team.slack.com/archives/CQUA4D5KJ/p1650468639024599?thread_ts=1650401626.612759&cid=CQUA4D5KJ) |
| - | - | - |
| ![Main](https://user-images.githubusercontent.com/62723358/164092906-dabc029a-72dd-472f-b998-ea4dc848c535.png) | ![PR](https://user-images.githubusercontent.com/62723358/164092903-6ddcc070-c47f-4491-b272-d5b42cb685b5.png) | ![Proposal](https://user-images.githubusercontent.com/62723358/164270690-72bc34dd-67bd-4f28-91d8-c25ed757ecf3.png) |

https://user-images.githubusercontent.com/62723358/164092587-6bfbcb75-693f-4f3f-8b88-4ab5a287792f.mov

| Before Click | After Click |
| - | - |
| ![Before](https://user-images.githubusercontent.com/62723358/164092591-94ea4930-45da-468b-b930-99257995e915.png) | ![After](https://user-images.githubusercontent.com/62723358/164092589-bdf71f13-0c55-4383-9baf-e466352c019b.png) | 


## Questions:

What is the easy way to get messages to re-appear after dismissing them? _They are no longer in browser storage, so I can't just delete their statuses like I could before._
